packages/test-kit/src/with-feature.ts

```diff
@@ -218,14 +225,22 @@ export function withFeature(withFeatureOptions: IWithFeatureOptions = {}) {
             if (defaultViewport) {
                 await page.setViewport(defaultViewport);
             }
-            pages.add(page);
-            page.on('pageerror', e => {
-                capturedErrors.push(e);
-                console.error(e);
-            });
+
+            (function handlePage(page: puppeteer.Page) {
```

Comment:
avoid overshadowing parameters. not sure how this is not blowing lint. :/
`subscribeToPage` might be a better name.